### PR TITLE
[Java] Fix Sequence unit tests

### PR DESCRIPTION
**Pull request by Batta32**

### CODE REVIEW

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/SequenceRecognizer.java [hunk 1]
```diff
@@ -77,42 +77,49 @@ public SequenceRecognizer(SequenceOptions options) {
 
     public static List<ModelResult> recognizePhoneNumber(String query, String culture, SequenceOptions options,
             Boolean fallbackToDefaultCulture) {
+        options = options != null ? options : SequenceOptions.None;
```
Comment (VictorGrycuk) [inline Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/SequenceRecognizer.java:80]: I know I added these changes, but can't this validation be added to a deeper or higher method, so it has to be validated once?

**Review by fran893 — APPROVED** [review]
LGTM

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BaseSequenceExtractor.java [hunk 2]
```diff
@@ -75,15 +75,14 @@ protected String getExtractType() {
...
-                    if (matchStream.anyMatch(o -> matchFunc.apply((Match)o))) {
+                    if (matchSource.keySet().stream().anyMatch(o -> matchFunc.apply(o))) {
                         Match srcMatch = (Match)matchSource.keySet().toArray()[0];
                         ExtractResult extResult = new ExtractResult();
 
                         extResult.setStart(start);
                         extResult.setLength(length);
                         extResult.setText(substr);
-                        extResult.setType(this.extractType);
+                        extResult.setType(this.getExtractType());
```
Comment (VictorGrycuk) [inline Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BaseSequenceExtractor.java]: Why this change? Why the class uses the accessor to get its own property?

File: Java/samples/simple-console/src/main/java/com/microsoft/recognizers/text/samples/simpleconsole/Sample.java [hunk 3]
```diff
@@ -17,7 +19,7 @@
 
 public class Sample {
 
-    private static final String defaultCulture = Culture.Spanish;
+    private static final String defaultCulture = Culture.English;
```
Comment (VictorGrycuk) [inline Java/samples/simple-console/src/main/java/com/microsoft/recognizers/text/samples/simpleconsole/Sample.java]: Should these changes be part of this PR?

**Review by VictorGrycuk — COMMENTED** [review]
I added a few comments

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/resources/ChineseIp.java [hunk 4]
```diff
@@ -14,17 +14,17 @@
 public class ChineseIp {
 
     public static final String Ipv4Regex = "({PhoneNumbersDefinitions.WordBoundariesRegex}(1\\d{2}|2[0-4]\\d|25[0-5]|0?[1-9]\\d|0{0,2}\\d)((\\.(1\\d{2}|2[0-4]\\d|25[0-5]|0?[1-9]\\d|0{0,2}\\d)){3}){PhoneNumbersDefinitions.EndWordBoundariesRegex})"
-            .replace("{PhoneNumbersDefinitions.WordBoundariesRegex}", PhoneNumbersDefinitions.WordBoundariesRegex)
-            .replace("{PhoneNumbersDefinitions.EndWordBoundariesRegex}", PhoneNumbersDefinitions.EndWordBoundariesRegex);
+            .replace("{PhoneNumbersDefinitions.WordBoundariesRegex}", BasePhoneNumbers.WordBoundariesRegex)
```
Comment (matiasroldan6) [inline Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/resources/ChineseIp.java:17]: Are these ok? I thought these were temporary solutions

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BaseSequenceExtractor.java (hunk 2 above)
Comment (matiasroldan6) [inline Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BaseSequenceExtractor.java]: here we change the access to the variable to access through its getter. Is it ok that we can access both the variable through the getter and directly? Generally, if there is a getter we shouldn't be able to access the variable directly, but I don't know about Java

Comment (VictorGrycuk) [reply Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BaseSequenceExtractor.java]: Same

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/SequenceRecognizer.java (hunk 1 above)
Comment (Batta32) [reply Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/SequenceRecognizer.java:80]: The problem here is that this method is `public`, if we add this in a higher method we can still have this issue. I'd prefer to leave this temporaly

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BaseSequenceExtractor.java (hunk 2 above)
Comment (Batta32) [reply Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BaseSequenceExtractor.java]: I'll double check

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/resources/ChineseIp.java (hunk 4 above)
Comment (Batta32) [reply Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/resources/ChineseIp.java:17]: Here we are fixing the generated resources.. IMO the problem is the resource-generator, but with these changes we are unblocking all.

Without these changes, the solution won't compile

File: Java/samples/simple-console/src/main/java/com/microsoft/recognizers/text/samples/simpleconsole/Sample.java (hunk 3 above)
Comment (Batta32) [reply Java/samples/simple-console/src/main/java/com/microsoft/recognizers/text/samples/simpleconsole/Sample.java]: This can be added in other PR, yes. I'll remove it

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BaseSequenceExtractor.java (hunk 2 above)
Comment (Batta32) [reply Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BaseSequenceExtractor.java]: +1 I'll double check

Comment (Batta32) [reply Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BaseSequenceExtractor.java]: Yep, that's correct, I rollbacked this. I'll push it asap

Comment (Batta32) [reply Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BaseSequenceExtractor.java]: Yep, that's correct, I rollbacked this. I'll push it asap

File: Java/samples/simple-console/src/main/java/com/microsoft/recognizers/text/samples/simpleconsole/Sample.java (hunk 3 above)
Comment (Batta32) [reply Java/samples/simple-console/src/main/java/com/microsoft/recognizers/text/samples/simpleconsole/Sample.java]: Done

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/SequenceRecognizer.java (hunk 1 above)
Comment (VictorGrycuk) [reply Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/SequenceRecognizer.java:80]: ack

**Comment by VictorGrycuk** [comment]
LGTM

**Review by VictorGrycuk — APPROVED** [review]

**Review by matiasroldan6 — APPROVED** [review]
